### ISSUE TITLE
Disable market pricechecking functionality

### ISSUE THREAD

**Issue by MikeLund** [Nuklon/Steam-Economy-Enhancer#41]
Is there any way to do this? I don't need your script to go through my 50+ pages of market listings, so it wastes HTTP requests and increases risk of being throttled by Valve.

(In general it'd be great if more stuff was able to be configured in your options dialogue. Thanks!)

**Comment by Nuklon** [comment]
This is not possible, but price checking isn't as resource hungry as you expect it to be. It only processes the items from the current page. It also adds a significant delay to avoid throttling.

**Comment by MikeLund** [comment]
Loading market pages:
It does a /market/mylistings?count=100&start=xxx for every page (me with 50+ pages takes a while to even display my listings)

Then when listings are shown:
First it does a /market/pricehistory/?appid=&market_hash_name= request for the next shown item.
Directly after it does a /market/itemordershistogram?language=english&currency=1&item_nameid=xxxxxx&two_factor=0 for this item.
And repeats this for page.


I would imagine that you wouldn't be able to easily remove the `mylistings` requests since the pagination is pretty baked in (and looks nice); but you should be able to add a toggle for price-checking if I'm not totally insane.

Completely understood though if you not interested in doing it though, thanks for the reply and feel free to close issue if so :)

**Comment by Nuklon** [comment]
I thought you meant inventory, so my reply doesn't make any sense...

Anyway, the /mylistings isn't throttled by Steam (hence no additional delay). 50+ pages will take some time, do you really have 5000+ items on the market? O_o 

The checking of the pricing is done with a delay to avoid Steam throttling. What's actually causing most throttle errors, opening the actual market listings page for the market id, is permanently stored to your disk in cache. Technically it's certainly possible to disable the price check, but do you actually encounter issues with it enabled?

**Comment by MikeLund** [comment]
Ah, ok. And yes I have several thousand items listed :)

I don't encounter any issues yet because of pricechecking, but point is that if I don't need that feature, it's cool to be able to be toggled off. Never know if Valve become more strict with throttling, apply the same throttle buckets to other more important requests, etc.

**Comment by Nuklon** [comment]
I'm going to put this on hold for now, quite some functionality depends on the price checking and until there's actually a need to remove this (for example, for throttling, which doesn't happen right now as SEE is already pretty throttle friendly).

**Comment by rluik** [comment]
The problem for me is that this feature gets on top of Steam Enhancer's lowest price in market listings so I can't see them. I know that I can hover  the colored square to see the lowest current listed price as a tooltip but that doesn't work on mobile (Firefox, Yandex for Android...).

**Comment by K-Boom79** [comment]
I have way too many cards on my inventory, and over 700 pages :D
Using together with enhanced steam freezes the page listing my inventory. Sad :(

**Comment by woctezuma** [comment]
> Never know if Valve become more strict with throttling

I revisit this issue after #165 and want to mention that there was an easy fix for the original poster.

Indeed, there are are few lines near the top of the script with `@include` keywords:

https://github.com/Nuklon/Steam-Economy-Enhancer/blob/c9a36827a898b03bc218f4d5b7bed017828e5e56/code.user.js#L9-L12

If you want to only use the script for inventory and trade offers, you can do so by removing the line mentioning the market:

https://github.com/Nuklon/Steam-Economy-Enhancer/blob/c9a36827a898b03bc218f4d5b7bed017828e5e56/code.user.js#L11
